sourcetracker/_sourcetracker.py

```diff
@@ -9,195 +9,203 @@
 # ----------------------------------------------------------------------------
 from __future__ import division
 
-import os
-from copy import copy
 import numpy as np
 from skbio.stats import subsample_counts
 import pandas as pd
 from functools import partial
-from biom.table import Table
-import glob
-from tempfile import TemporaryDirectory
 
 
-def parse_mapping_file(mf_lines):
-    """Parse lines from a mapping file into a nested dictionary.
+def check_and_correct_data(feature_table, cfc):
+    '''Check and correct problems in the `feature_table`.
 
-    This function copies qiime.parse_mapping_file_to_dict, but does _not_ allow
-    commented lines at the top of a mapping file.
+    Parameters
+    ----------
+    feature_table : pd.DataFrame
+        Contingency table with rows, columns = samples, features.
+    cfc : boolean
+        Fractional counts cause errors in indexing. Uses np.ceil to remove
+        them if True.
+
+    Returns
+    -------
+    feature_table : pd.DataFrame
+        Corrected feature table if changes have been made or original table if
+        no changes occurred. Will automatically be cast as dtype=np.int64
+
+    Raises
+    ------
+    ValueError
+        If data contains values that Gibb's sampling cannot handle.
+    '''
+    _ft = feature_table.copy()
+    if _ft.isnull().values.sum() != 0:
+        raise ValueError('One or more values in the feature table is '
+                         'non-numeric. This will cause propagating error and '
+                         'must be removed.')
+    if not (np.ceil(_ft.values) == _ft.values).all():
+        if cfc is True:
+            _ft.iloc[:, :] = np.ceil(_ft.values)
```

Comment:
@lkursell @johnchase 

Cases where the input feature table has contained nans and floats have caused errors several times. The reason we have the ceil call in there is to handle cases where the user has collapsed sources or sinks with `mean`.  This was a frequent use case for @lkursell and he wanted the one-line API call to include a way to handle this. Happy to discuss changing this behavior, but need their input.